### PR TITLE
Feat/space fk

### PR DESCRIPTION
**Pull request by hetjagani**

### CODE REVIEW

File: service/core/action/category/update.go [hunk 1]
```diff
@@ -70,13 +74,20 @@ func update(w http.ResponseWriter, r *http.Request) {
...
 		MediumID:    category.MediumID,
-	}).Preload("Medium").First(&result)
+	}).Error
+
+	if err != nil {
+		errorx.Render(w, errorx.Parser(errorx.DBError()))
+		return
+	}
+
+	config.DB.Model(&model.Category{}).Preload("Medium").First(&result)
```
Comment (dmonark) [inline service/core/action/category/update.go]: @shreeharsha-factly check this

Comment (hetjagani) [reply service/core/action/category/update.go]: This is to handle Foreign Key constrains error during update same as in create.go

Comment (shreeharsha-factly) [reply service/core/action/category/update.go]: Even if u take error after **First** query still it handles FK constrain error. So u can directly add first query after update & then take error.

File: service/core/action/category/update.go [hunk 2]
```diff
@@ -80,14 +80,12 @@ func update(w http.ResponseWriter, r *http.Request) {
...
-	}).Error
+	}).Preload("Medium").First(&result).Error
 
 	if err != nil {
 		errorx.Render(w, errorx.Parser(errorx.DBError()))
 		return
 	}
 
-	config.DB.Model(&model.Category{}).Preload("Medium").First(&result)
-
```
Comment (hetjagani) [inline service/core/action/category/update.go]: Made the change. @shreeharsha-factly is it ok?

**Review by shreeharsha-factly — APPROVED** [review]

**Pull request merged**